### PR TITLE
enhance: [2.4] new knowhere param for range_search

### PR DESCRIPTION
**Pull request by alwayslove2013**
Cherry-pick from master
pr: #34686
issue: #34685

### CODE REVIEW

File: internal/core/src/index/VectorMemIndex.cpp [hunk 1]
```diff
@@ -600,6 +600,10 @@ VectorMemIndex<T>::Query(const DatasetPtr dataset,
                                       search_conf[RANGE_FILTER],
                                       GetMetricType());
             }
+            // `range_search_k` is only used as one of the conditions for early termination.
```
Comment (cydrain) [inline internal/core/src/index/VectorMemIndex.cpp:603]: you can use existed param "k" for range search

Comment (alwayslove2013) [reply internal/core/src/index/VectorMemIndex.cpp:603]: The meanings are not quite the same
1. `range_search_k` does not promise to return exactly `k` results.
2. `k` use `uint`, but `range_search_k` wanna use `-1` to preserve the possibility of returning all results without k limitation.

Comment (cydrain) [reply internal/core/src/index/VectorMemIndex.cpp:603]: ok

**Pull request merged**